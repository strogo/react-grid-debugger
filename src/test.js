import React from 'react'
import renderer from 'react-test-renderer'

import GridDebugger from '../src/index'

it('renders <GridDebugger /> with single value props correctly', () => {
  const tree = renderer.create(
    <GridDebugger 
      show
      maxWidth={'1000px'}
      numCols={12}
      gutter={'4rem'}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <GridDebugger /> with object props correctly', () => {
  const tree = renderer.create(
    <GridDebugger 
      show
      maxWidth={{ 320: '300px', 580: 'none', 800: '1000px' }}
      numCols={{ 320: 1, 680: 2, 800: 12 }}
      gutter={{ 320: '2rem', 680: '40px', 800: '50px' }}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <GridDebugger /> with array props correctly', () => {
  const tree = renderer.create(
    <GridDebugger
      show
      maxWidth={['280px', '600px', '1000px']}
      numCols={[1, 4, 12]}
      gutter={['2rem', '40px', '50px', '1000px']}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
