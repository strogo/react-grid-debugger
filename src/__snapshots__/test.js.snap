// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <GridDebugger /> with array props correctly 1`] = `
<div
  className="css-dcj038"
  cols={
    Object {
      "map": Object {
        "0": 1,
        "1025": 12,
        "737": 4,
      },
      "singleValue": 12,
    }
  }
>
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
</div>
`;

exports[`renders <GridDebugger /> with object props correctly 1`] = `
<div
  className="css-16zwufx"
  cols={
    Object {
      "map": Object {
        "320": 1,
        "680": 2,
        "800": 12,
      },
      "singleValue": 12,
    }
  }
>
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
</div>
`;

exports[`renders <GridDebugger /> with single value props correctly 1`] = `
<div
  className="css-1cbjbb6"
  cols={
    Object {
      "singleValue": 12,
    }
  }
>
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
  <div
    className="css-1tjnv1f"
  />
</div>
`;
